docs/elasticsearch-spec.asciidoc

```diff
@@ -101,42 +144,94 @@ $ openssl req -x509 -newkey rsa:4096 -keyout tls.key -out tls.crt -days 365 -nod
-https://www.elastic.co/guide/en/elasticsearch/reference/current/vm-max-map-count.html
 
-Optionally, you can select a different type of file system implementation for the storage. Here you can find information about possible options:
-https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules-store.html
+More common is setting resource requests and limits, which is covered in the section <<{p}-resource-requests-limits>>.
+
+You may want to also install additional plugins, which is described in the <<{p}-init-containers-plugin-downloads>> section.
+
+You may want to set environment variables to configure Elasticsearch, for instance to set the minimum and maximum heap size to `2g` and `4g` respectively you may modify the environment variables of the `elasticsearch` container as follows:
 
 [source,yaml]
 ----
-spec:
-  nodes:
-  - nodeCount: 3
-    config:
-      index.store.type: niofs
+    podTemplate:
+      spec:
+        containers:
+        - name: elasticsearch
+          env:
+          - name: ES_JAVA_OPTS
+            value: "-Xms2g -Xmx4g"
+----
+
+
+You can also refer to the Kubernetes documentation here for more information on pod templates:
+https://kubernetes.io/docs/concepts/workloads/pods/pod-overview/#pod-templates[Pod Templates Overview]
+
+https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.15/#podtemplatespec-v1-core[Pod Template Spec API Reference]
+
+
+[id="{p}-resource-requests-limits"]
+=== Resource Requests and Limits
+
+Resource requests and limits define how much RAM and CPU each individual container requires and how much they are allowed to use. You may want to set these in addition to the heap size as mentioned , though by default we will set the heap size to half of the container memory limit.
+
+To set the minimum and maximum memory of the `elasticsearch` container, as well as the minimum and maximum CPU resources, you would modify the pod template in your Elasticsearch spec as follows.
+[source,yaml]
+----
+    podTemplate:
+      spec:
+        containers:
+        - name: elasticsearch
+          resources:
+            requests:
+              memory: "16Gi"
+              cpu: "250m"
+            limits:
+              memory: "32Gi"
+              cpu: "4"
 ----
 
+The https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/[Kubernetes documentation] has more detail on managing compute resources for containers.
+
+[id="{p}-init-containers-plugin-downloads"]
+=== Init containers for plugin downloads
+
+To install a custom plugin, you can install them before the Elasticsearch container starts with an initContainer. For example:
+
+[source,yaml]
+----
+  - podTemplate:
+      spec:
+        initContainers:
+        - name: install-plugins
+          image: docker.elastic.co/elasticsearch/elasticsearch:7.1.0"
+          command:
+          - sh
+          - -c
+          - |
+            bin/elasticsearch-plugin install --batch repository-gcs
+----
+
```

Comment:
Would be nice here to document what init containers inherit by default: https://github.com/elastic/cloud-on-k8s/blob/master/operators/pkg/controller/common/defaults/pod_template.go#L229